### ISSUE TITLE
Support for netbox 3.6.0

### ISSUE THREAD

**Issue by Kaelnor** [bb-Ricardo/netbox-sync#347]
Hi,

Some breaking changes happened to the netbox API with 3.6.0 (https://docs.netbox.dev/en/stable/release-notes/version-3.6/#breaking-changes). Most notably, the rename from `device_role` to `role` in the device model. `device_role` is now read-only.

Both fields are identical, queried from a device:

``` json
            "role": {
                "id": 1,
                "url": "http://netbox-staging/api/dcim/device-roles/1/",
                "display": "Server",
                "name": "Server",
                "slug": "server"
            },
            "device_role": {
                "id": 1,
                "url": "http://netbox-staging/api/dcim/device-roles/1/",
                "display": "Server",
                "name": "Server",
                "slug": "server"
            },
```

For now, netbox-sync only tries to send `device_role` when creating a new device:

```
2023-09-20 10:01:39,077 - ERROR: NetBox returned: POST /api/dcim/devices/ Bad Request
2023-09-20 10:01:39,077 - ERROR: NetBox returned body: {'role': ['This field is required.']}
2023-09-20 10:01:39,077 - ERROR: Request Failed for device. Used data: {'tags': [{'name': 'NetBox-synced'}, {'name': 'Source: REDACTED'}], 'name': 'REDACTED', 'device_type': 110, 'site': 6, 'cluster': 12, 'status': 'active', 'serial': 'REDACTED', 'platform': 1, 'custom_fields': {'vcsa_host_cpu_cores': '16 Intel(R) Xeon(R) CPU E5-2650 v2 @ 2.60GHz', 'vcsa_host_memory': '128 GB'}, 'device_role': 1, 'primary_ip4': 4136}
```

I don't know how you'd like to handle this. Maybe setting a new attribute `role` in NBDevice class as a duplicate of the existing `device_role` and send both ? 
https://github.com/bb-Ricardo/netbox-sync/blob/b9e236df43c463d4d09a8afeeb6d69f3f8ee62aa/module/netbox/object_classes.py#L1503-L1519

That would be the easiest to stay compatible with <3.6.0 if the netbox API gracefully handles POSTing additional and read-only fields.

**Comment by bb-Ricardo** [comment]
Hi, thank you for reporting this.

Have also seen that there is a new 3.6.1 out. Maybe I can just whip up something quickly to mitigate this issue.

**Comment by bb-Ricardo** [comment]
Hi,

added a fix to the new branch "fix/support-netbox-3.6". Can you try this one out and see if this resolves the issue.

I tested it locally and it seems to work as intended. The longest was spent on setting up a Minikube environment with the latest Postgres and NetBox versions. I assume not many people are running NetBox on k8s.

**Comment by Kaelnor** [comment]
Thanks @bb-Ricardo for the fix, that was quick indeed ! 👍🏼 

After testing on 3.6.1, all looks good. The synchronization creates missing hosts correctly. I can't test on netbox <3.6.0 though.

**Comment by bb-Ricardo** [comment]
Thank you for testing it.

**Comment by bb-Ricardo** [comment]
Fixed in release v1.5.1
